language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/InterfaceClass.scala

```diff
@@ -102,35 +103,45 @@ object InterfaceClass extends StrictLogging {
       interfaceName: ClassName,
       choiceNames: Set[ChoiceName],
       interfaceViewTypeName: ClassName,
-  ): TypeSpec = TypeSpec
-    .classBuilder(companionClassName)
-    .superclass(
-      ParameterizedTypeName
-        .get(ClassName get classOf[InterfaceCompanion[_, _]], interfaceName, interfaceViewTypeName)
-    )
-    .addModifiers(Modifier.FINAL, Modifier.PUBLIC, Modifier.STATIC)
-    .addMethod {
-      MethodSpec
-        .constructorBuilder()
-        // intentionally package-private
-        .addStatement(
-          "super($T.$N, $T.$L(), $T.of($L))",
-          interfaceName,
-          ClassGenUtils.templateIdFieldName,
-          interfaceViewTypeName,
-          "valueDecoder",
-          classOf[java.util.List[_]],
-          CodeBlock
-            .join(
-              choiceNames
-                .map(choiceName => CodeBlock.of("$N", toChoiceNameField(choiceName)))
-                .asJava,
-              ",$W",
-            ),
-        )
-        .build()
-    }
-    .build()
+  ): TypeSpec = {
+    val contractIdClassName = ClassName bestGuess "ContractId"
+    TypeSpec
+      .classBuilder(companionClassName)
+      .superclass(
+        ParameterizedTypeName
+          .get(
+            ClassName get classOf[InterfaceCompanion[_, _, _]],
+            interfaceName,
+            contractIdClassName,
+            interfaceViewTypeName,
+          )
+      )
+      .addModifiers(Modifier.FINAL, Modifier.PUBLIC, Modifier.STATIC)
+      .addMethod {
+        MethodSpec
+          .constructorBuilder()
+          // intentionally package-private
+          .addStatement(
+            "super($>$Z$S, $T.$N, $T::new, $T.$L(), $T.of($L))$<$Z",
+            interfaceName,
+            interfaceName,
+            ClassGenUtils.templateIdFieldName,
+            contractIdClassName,
+            interfaceViewTypeName,
+            "valueDecoder",
+            classOf[List[_]],
```

Comment:
This is definitely a reference to `scala.List` now; I don't know why it would build, but please revert for sanity's sake.